client/src/features/session/components/StartNewSession.tsx

```diff
@@ -158,21 +158,29 @@ function BackButton() {
   const projectUrl = Url.get(Url.pages.project, projectUrlData);
 
   const location = useLocation<LocationState | undefined>();
+
   const { from, filePath } = location.state ?? {};
-  const backUrl = from ?? projectUrl;
+  const searchParams = getSearchParams();
+  const fromLandingParam = (searchParams as { fromLanding?: string })
+    .fromLanding;
+  const fromLanding = location.state?.fromLanding || fromLandingParam || false;
```

Comment:
Avoid the untyped `getSearchParams()`:

```suggestion
 const searchParams = useMemo(
    () => new URLSearchParams(location.search),
    [location.search]
  );
  const fromLandingParam = !!searchParams.get("fromLanding");
  const fromLanding = location.state?.fromLanding || fromLandingParam;
```